### PR TITLE
Recovery tests cleanup

### PR DESCRIPTION
**Pull request by zalokhan**
### Recovery test cleanup
- Cleanup AbstractIT class. Addressing review comments.
- By default only running unit tests.
- Running integration tests only in travis or by running them on demand. Fixing #580 
Note: Disabling StreamIT #520 
### NettyClientRouter fixing leaking threads
- Graceful shutdown of netty client routers threads.

### CODE REVIEW

**Comment by slfritchie** [comment]
Zeeshan, the new Travis execution of the IT-style tests failed: https://travis-ci.org/CorfuDB/CorfuDB/builds/228535942#L2195.  Does the argument with wildcards need to be quoted?

**Comment by slfritchie** [comment]
Using `./mvnw '-Dtest=*IT' -DfailIfNoTests=false test` seems to work, at least with the C shell that my account uses.  However, both the `SealIT` and the `ServerRestartIT` tests fail to start a Corfu server on my OS X machine, hrm.

**Comment by zalokhan** [comment]
@slfritchie Yes, I am trying to fix this.
So I have modified SealIT to extend and use the new integration framework.
I have disabled StreamIT for now as it is an outdated test.
I'll try your fix

**Comment by zalokhan** [comment]
I have made a couple of changes to the pom.xml to verify integration test. 
Kindly review this carefully.

File: infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java [hunk 1]
```diff
@@ -474,6 +474,4 @@ private void handleAllocation(CorfuPayloadMsg<TokenRequest> msg,
                         token,
                         backPointerMap.build())));
     }
-
-    private static final int globalTokenBatchSize = 100;
```
Comment (dahliamalkhi) [inline infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java]: thanks!

File: test/src/test/java/org/corfudb/integration/StreamIT.java [hunk 2]
```diff
@@ -31,10 +31,12 @@ static public void getLayoutServers() throws Exception {
         layoutServers = (String) properties.get("layoutServers");
     }
 
-    @Test
+//    @Test
```
Comment (dahliamalkhi) [inline test/src/test/java/org/corfudb/integration/StreamIT.java]: is this intentional?

Comment (zalokhan) [inline test/src/test/java/org/corfudb/integration/StreamIT.java]: Yes, The logic needs to be changed. #520

**Comment by zalokhan** [comment]
@slfritchie regarding the increased running time, I have modified the pom so that these tests are not run on every mvn clean install. Only on mvn verify which is run by travis now.
Do we still want to reduce the iterations ?
@medhavidhawan @Maithem I would like your input on this too.

**Comment by slfritchie** [comment]
@zalokhan This patch roughly doubles the time required for a travis-ci run. Is there additional value in running beyond 3 iterations: client restart, server restart, both restart?

**Comment by zalokhan** [comment]
Okay, I will decrease the iterations and make it configurable.

**Comment by corfudb-performance** [comment]
Results automatically generated by _CorfuDB Benchmark Framework_ to assess the performance of this pull request for commit **21e4bf2**.

*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 1 threads, Memory mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 5 threads, Memory mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 10 threads, Memory mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 1 threads, Memory mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 5 threads, Memory mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 10 threads, Memory mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 10 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 10 threads, Disk mode
 
 An interactive dashboard with Pull Request Performance Metrics for ALL cluster types and numbers of threads in run, is available at:
 [Pull Request #586 Graphs ](https://snapshot.raintank.io/dashboard/snapshot/Vvc8YoKkJX8ypkMKtwEd0cgHT8oHozas)

File: runtime/src/main/java/org/corfudb/runtime/clients/NettyClientRouter.java [hunk 3]
```diff
@@ -156,7 +156,7 @@
      * Are we connected?
      */
     @Getter
-    Boolean connected_p;
+    volatile Boolean connected_p;
```
Comment (no2chem) [inline runtime/src/main/java/org/corfudb/runtime/clients/NettyClientRouter.java]: to continue with the name issues, let's rename connected_p -> connected

**Comment by corfudb-performance** [comment]
Results automatically generated by _CorfuDB Benchmark Framework_ to assess the performance of this pull request for commit **c89b09f**.

*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 1 threads, Memory mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 5 threads, Memory mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 10 threads, Memory mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 1 threads, Memory mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 5 threads, Memory mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 10 threads, Memory mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 10 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 10 threads, Disk mode
 
 An interactive dashboard with Pull Request Performance Metrics for ALL cluster types and numbers of threads in run, is available at:
 [Pull Request #586 Graphs ](https://snapshot.raintank.io/dashboard/snapshot/2WFzgiaCP1WDgP4K7yGTKNIr1A9L5S3I)

File: runtime/src/main/java/org/corfudb/runtime/clients/NettyClientRouter.java (hunk 3 above)
Comment (zalokhan) [inline runtime/src/main/java/org/corfudb/runtime/clients/NettyClientRouter.java]: Oh yes sorry, I intended to commit that change too.

**Comment by zalokhan** [comment]
Done.

**Pull request merged**